celery/app/base.py

```diff
@@ -1149,24 +1151,25 @@ def current_worker_task(self):
         """
         return get_current_worker_task()
 
-    @cached_property
+    @property
     def oid(self):
         """Universally unique identifier for this app."""
-        # since 4.0: thread.get_ident() is not included when
-        # generating the process id.  This is due to how the RPC
-        # backend now dedicates a single thread to receive results,
-        # which would not work if each thread has a separate id.
```

Comment:
I'd like to see some validation that this comment is not longer relevant. The `celery.backend.asynchronous.BaseResultConsumer` class is used fairly broadly now and it sounds like messing this up would result in us losing results all over the place. If we have to fix it, I figure we can pass a specific OID down to the `RPCBackend` rather than allowing it to access the `app.oid` like we currently do in:

https://github.com/celery/celery/blob/7c3da03a07882ca86b801ad78dd509a67cba60af/celery/backends/rpc.py#L339-L342